src/test/compile-fail/feature-gate-overlapping_marker_traits.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+trait MyMarker {}
```

Comment:
can you add a comment explaining that this test is targeting the absnce of a feature-gate?

Also, I think I'd prefer to have the `T: Eq` vs `T: Debug` example (or whatever) rather than this one, since this also works with specialization.